### ISSUE TITLE
GET_LOCAL_IP: Can't resolve host

### ISSUE THREAD

**Issue by Hooch76** [SukramJ/hahomematic#1387]
### I agree to the following

- [X] I have read the [documentation](https://github.com/danielperna84/custom_homematic#custom_homematic) right now
- [X] I have read the [FAQs](https://github.com/danielperna84/custom_homematic#frequently-asked-questions) right now
- [X] I am aware of the latest [release notes](https://github.com/danielperna84/custom_homematic/releases)
- [X] The Backend (CCU/Homegear/...) is working as expected and connected devices are controllable by its own UI.
- [X] I am running the latest version of the custom_component (and Home Assistant)

### The problem

Trying to set up / configure the Home Assistant integration, but it always fails with the following error:

```
2024-01-17 22:21:41.934 WARNING (SyncWorker_5) [hahomematic.central] GET_LOCAL_IP: Can't resolve host for 127.0.0.1:8200
2024-01-17 22:21:41.966 ERROR (MainThread) [hahomematic.client.xml_rpc] OSError on temporary_instance-HmIP-RF: (-2, 'Name does not resolve')
2024-01-17 22:21:41.967 WARNING (MainThread) [hahomematic.central] OSError on temporary_instance-HmIP-RF: (-2, 'Name does not resolve')
```
It does not matter if I use 127.0.0.1, the real IP of the debmatic CCU or a hostname that can be resolved. 
Note that I am using a custom port (8200).

### What version of HomematicIP (local) has the issue?

1.54.0

### What was the last working version of HomematicIP (local)?

Never used it before

### What type of installation are you running?

Home Assistant Container

### What type of installation are you running for your homematic backend?

Debmatic

### Which version of your homematic backend are you running?

3.73.9.112

### What hardware are you running for your system?

Intel x64, Debian

### Which config details do you use

- [X] Username and password
- [ ] TLS
- [ ] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [X] Homematic IP
- [X] Homematic (Bidcos-RF)
- [ ] Groups (Heating-Group)
- [ ] BidCos-Wired (HM-Wired)

### Diagnostics information

2024-01-17 22:21:41.934 WARNING (SyncWorker_5) [hahomematic.central] GET_LOCAL_IP: Can't resolve host for 127.0.0.1:8200
2024-01-17 22:21:41.966 ERROR (MainThread) [hahomematic.client.xml_rpc] OSError on temporary_instance-HmIP-RF: (-2, 'Name does not resolve')
2024-01-17 22:21:41.967 WARNING (MainThread) [hahomematic.central] OSError on temporary_instance-HmIP-RF: (-2, 'Name does not resolve')

**Comment by Hooch76** [comment]
I have no idea about Python, but I found https://github.com/danielperna84/hahomematic/blob/abac92f5138e84b27fa98ce63b29c46b033b632c/hahomematic/central/__init__.py#L504
 which should be the part where the error is thrown, right?

So I created the following script and ran in in the docker container where HomeAssistant is running:

```
import socket
hostname = "127.0.0.1"

try:
    ip_address = socket.gethostbyname(hostname)
    print(f"Die IP-Adresse von {hostname} lautet: {ip_address}")
except socket.gaierror:

    print("Hostname konnte nicht aufgelöst werden.")
```

And guess what? It worked for 127.0.0.1, the real IP address and the hostname. All 3 were tried when trying to configure the HomeAssistant integration. 
I don't know if it helps though :-)
